### ISSUE TITLE
ItemList disabled items trigger selection and activation signals on mouse

### ISSUE THREAD

**Issue by YuriSizov** [godotengine/godot#37277]
**Godot version:**
3.2.1.stable

**OS/device including version:**
Windows 10

**Issue description:**
According to `ItemList` documentation setting an item disabled means:
> Disabled items cannot be selected and do not trigger activation signals (when double-clicking or pressing Enter).

However, when clicked with mouse, they can still trigger `item_selected` and `item_activated` signals. Visually such items appear disabled and they cannot be selected with keyboard either: pressing arrow keys does not move the selection if the next item is disabled. But on clicks they do react.

Selection itself never changes, and signals can be filtered with `is_item_disabled`. So, this may be an error in classref. The only similar issue I have found was related to 2.1 (#5683), which resulted in fixing item behaviour.

**Steps to reproduce:**
1. Create a UI scene with an `ItemList` node.
2. Create a similar script to fill `ItemList` with items and disable one:
```swift
extends Control

onready var item_list : ItemList = $ItemList

func _ready() -> void:
	item_list.clear()
	item_list.add_item("test item 1")
	item_list.add_item("test item 2")
	item_list.add_item("test item 3")
	item_list.add_item("test item 4")
	item_list.add_item("test item 5")
	
	item_list.set_item_disabled(2, true)
	
	item_list.connect("item_activated", self, "_on_item_activated")
	item_list.connect("item_selected", self, "_on_item_selected")

func _on_item_activated(item_index : int) -> void:
	print("item #" + str(item_index) + " was activated")

func _on_item_selected(item_index : int) -> void:
	print("item #" + str(item_index) + " was selected")
```
3. Run the project, clicking on the disabled item still produces prints:
>item #2 was selected
item #2 was activated

**Minimal reproduction project:**
[bug-itemlist-disabled-activation.zip](https://github.com/godotengine/godot/files/4375668/bug-itemlist-disabled-activation.zip)

**Comment by KoBeWi** [comment]
Fixed by #60123

**Comment by goblinJoel** [comment]
Any chance of getting the fix in 3.x as well?

**Comment by TaphouseGames** [comment]
Seems like this is an issue again in godot 3.5

**Comment by KoBeWi** [comment]
The fix was never backported, it's only in 4.0.
